userspace/libsinsp/test/filterchecks/k8s.cpp

```diff
@@ -0,0 +1,192 @@
+	ASSERT_TRUE(field_exists(evt, "k8s.pod.id"));
+	ASSERT_TRUE(field_exists(evt, "k8s.pod.label.one.second.third"));
+	ASSERT_TRUE(field_exists(evt, "k8s.pod.labels"));
+	ASSERT_TRUE(field_exists(evt, "k8s.pod.ip"));
+	ASSERT_TRUE(field_exists(evt, "k8s.pod.cni.json"));
+	ASSERT_TRUE(field_exists(evt, "k8s.ns.name"));
+
+	ASSERT_TRUE(field_exists(evt, "k8s.rc.name"));
+	ASSERT_TRUE(field_exists(evt, "k8s.rc.id"));
+	ASSERT_TRUE(field_exists(evt, "k8s.rc.label.one"));
+	ASSERT_TRUE(field_exists(evt, "k8s.rc.labels"));
+	ASSERT_TRUE(field_exists(evt, "k8s.svc.name"));
+	ASSERT_TRUE(field_exists(evt, "k8s.svc.id"));
+	ASSERT_TRUE(field_exists(evt, "k8s.svc.label.one"));
+	ASSERT_TRUE(field_exists(evt, "k8s.svc.labels"));
+	ASSERT_TRUE(field_exists(evt, "k8s.ns.id"));
+	ASSERT_TRUE(field_exists(evt, "k8s.ns.label.one"));
+	ASSERT_TRUE(field_exists(evt, "k8s.ns.labels"));
+	ASSERT_TRUE(field_exists(evt, "k8s.rs.name"));
+	ASSERT_TRUE(field_exists(evt, "k8s.rs.id"));
+	ASSERT_TRUE(field_exists(evt, "k8s.rs.label.one"));
+	ASSERT_TRUE(field_exists(evt, "k8s.rs.labels"));
+	ASSERT_TRUE(field_exists(evt, "k8s.deployment.name"));
+	ASSERT_TRUE(field_exists(evt, "k8s.deployment.id"));
+	ASSERT_TRUE(field_exists(evt, "k8s.deployment.label.one"));
+	ASSERT_TRUE(field_exists(evt, "k8s.deployment.labels"));
+
+	// There are no containers in the container manager, so there shouldn't be values
+	ASSERT_FALSE(field_has_value(evt, "k8s.pod.name"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.pod.id"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.pod.label.one"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.pod.labels"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.pod.ip"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.pod.cni.json"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.ns.name"));
+
+	// These fields will always exist but without a value, they are deprecated
+	ASSERT_FALSE(field_has_value(evt, "k8s.rc.name"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.rc.id"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.rc.label.one"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.rc.labels"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.svc.name"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.svc.id"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.svc.label.one"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.svc.labels"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.ns.id"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.ns.label.one"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.ns.labels"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.rs.name"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.rs.id"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.rs.label.one"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.rs.labels"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.deployment.name"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.deployment.id"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.deployment.label.one"));
+	ASSERT_FALSE(field_has_value(evt, "k8s.deployment.labels"));
+}
+
+TEST_F(sinsp_with_test_input, K8S_FILTER_check_fields_value)
```

Comment:
These are all ok, thank you! Will help making them slightly more realistic via mocking API responses, will tag you then @Andreagit97 .

Reply:
ok the PR is up https://github.com/falcosecurity/libs/pull/1544 once we adjust pod id we can update those tests.